woodwork/deserialize_accessor.py

```diff
@@ -0,0 +1,176 @@
+import json
+import os
+import tarfile
+import tempfile
+import warnings
+from itertools import zip_longest
+from pathlib import Path
+
+import pandas as pd
+
+import woodwork as ww
+from woodwork.exceptions import OutdatedSchemaWarning, UpgradeSchemaWarning
+from woodwork.s3_utils import get_transport_params, use_smartopen
+from woodwork.serialize_accessor import FORMATS, SCHEMA_VERSION
+from woodwork.utils import _is_s3, _is_url, import_or_raise
+
+
+def read_table_typing_information(path):
+    '''Read Woodwork typing information from disk, S3 path, or URL.
+
+        Args:
+            path (str): Location on disk, S3 path, or URL to read `table_description.json`.
+
+        Returns:
+            description (dict) : Woodwork typing information
```

Comment:
I don't think we need to name the return object so this could just be `dict: Woodwork typing information dictionary`